infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServerNode.java

```diff
@@ -1,82 +1,70 @@
+ */
 @Slf4j
 public class CorfuInterClusterReplicationServerNode implements AutoCloseable {
 
     @Getter
     private final ServerContext serverContext;
 
-    @Getter
-    private final Map<Class, AbstractServer> serverMap;
-
-    @Getter
-    private final LogReplicationServerRouter router;
-
-    // This flag makes the closing of the CorfuServer idempotent.
-    private final AtomicBoolean close;
-
-    private LogReplicationServer logReplicationServer;
-
     private ScheduledExecutorService logReplicationServerRunner;
 
     // Error code required to detect an ungraceful shutdown.
     private static final int EXIT_ERROR_CODE = 100;
 
+    // this flag makes the start/stop operation of transportLayerServer idempotent
+    @Getter
+    private AtomicBoolean serverStarted;
+
+    private final LogReplicationClientServerRouter router;
+
     /**
-     * Corfu Server initialization.
+     * Log Replication Server initialization.
      *
      * @param serverContext Initialized Server Context
-     * @param logReplicationServer Replication Server which processes incoming requests
+     * @param router Interface between LogReplication and the transport layer
      */
     public CorfuInterClusterReplicationServerNode(@Nonnull ServerContext serverContext,
-        LogReplicationServer logReplicationServer) {
+                                                  LogReplicationClientServerRouter router) {
 
         this.serverContext = serverContext;
+        this.router = router;
+        this.serverStarted = new AtomicBoolean(false);
+        startServer();
+    }
 
-        this.logReplicationServer = logReplicationServer;
-
-        this.serverMap = ImmutableMap.<Class, AbstractServer>builder()
-            .put(BaseServer.class, new BaseServer(serverContext))
-            .put(LogReplicationServer.class, logReplicationServer)
-            .build();
-
-        this.close = new AtomicBoolean(false);
-        this.router = new LogReplicationServerRouter(new ArrayList<>(serverMap.values()));
-        this.serverContext.setServerRouter(router);
-
-        logReplicationServerRunner = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
-            .setNameFormat("replication-server-runner").build());
-
-        // Start and listen to the server
-        logReplicationServerRunner.submit(this::startAndListen);
+    public void startServer() {
+        if (!serverStarted.get()) {
+            logReplicationServerRunner = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
+                    .setNameFormat("replication-server-runner").build());
+            // Start and listen to the server
+            logReplicationServerRunner.submit(() -> this.startAndListen());
+        } else {
+            log.info("Server transport adapter is already running. Updating the router information");
+        }
     }
 
     /**
      * Wait on Corfu Server Channel until it closes.
```

Comment:
little confuse here. Is it starting the transport server like grpc (in our case)? The naming/comments look like LR server and corfu server. Please clarify.

Reply:
yes, it is the transport layer server.
Updated the comment to 

> Wait on the transport frameworks's server until it is shutdown.